data/redis.go

```diff
@@ -80,14 +80,35 @@ func (r *RedisConnector) Id() string {
 
 // connectTask is the function that tries to establish a Redis connection (and pings to verify).
 func (r *RedisConnector) connectTask(ctx context.Context) error {
-	options := &redis.Options{
-		Addr:         r.cfg.Addr,
-		Password:     r.cfg.Password,
-		DB:           r.cfg.DB,
-		PoolSize:     r.cfg.ConnPoolSize,
-		DialTimeout:  r.initTimeout,
-		ReadTimeout:  r.getTimeout,
-		WriteTimeout: r.setTimeout,
+	var options *redis.Options
+	var err error
+
+	// URI is preferred and default connection behavior
+	if r.cfg.URI != "" {
+		r.logger.Debug().Str("uri", util.RedactEndpoint(r.cfg.URI)).Msg("attempting to connect to Redis using URI")
+		options, err = redis.ParseURL(r.cfg.URI)
+		if err != nil {
+			return fmt.Errorf("failed to parse Redis URI: %w", err)
+		}
+
+		options.DialTimeout = r.initTimeout
+		options.ReadTimeout = r.getTimeout
+		options.WriteTimeout = r.setTimeout
```

Comment:
do you need to set all this or can you rely on the undelrying redis lib to set all this up using the URI?

Reply:
seems like `redis-go` already has default options for all three [here](https://pkg.go.dev/github.com/redis/go-redis/v9#Options) when parsing the URI; but I don't think it is something you can setup as part of the URI?

we also have default values for them in erpc [here](https://github.com/erpc/erpc/blob/main/common/defaults.go#L655)

Reply:
I need to test if this is possible or not, but would this be more readable?

`redis://user:secret@cache.example.com:6379/0?pool_size=20&dial_timeout=4s&read_timeout=2s&write_timeout=2s`

Reply:
Now with the URI support probably you only want to do these only if `r.setTimeout > 0` and actually set...